lib/system/environ/validate.go

```diff
@@ -45,9 +49,21 @@ func ValidateInstall(env *localenv.LocalEnvironment) func() error {
 			// Operational state directory requirements are advisory
 			log.WithError(err).Warn("Failed to validate state directory requirements.")
 		}
-		// FIXME: validate enough of directory contents to make an educated decision automatically
-		if err := validateDirectoryEmpty(stateDir); err != nil {
-			return trace.Wrap(err)
+		if err := validateNoActiveService(stateDir); err != nil {
+			log.WithError(err).Warn("Failed to determine if service is not active.")
+			return trace.BadParameter("detected an active installer service in %v, "+
+				"please resume the agent with `gravity resume` or "+
+				"clean it up using `gravity leave --force` before proceeding "+
+				"(see https://gravitational.com/gravity/docs/cluster/#deleting-a-cluster for more details)",
+				stateDir)
+		}
+		if err := validateNoActiveOperation(stateDir); err != nil {
+			log.WithError(err).Warn("Failed to detect an active operation.")
```

Comment:
I don't want to be pedantic but this should probably only be logged in case `!trace.IsAlreadyExists(err)` :) Same for the if above.